integration/rpc_test.go

```diff
@@ -229,27 +230,62 @@ func TestGetStateRPC(t *testing.T) {
 	ctx, ctxCancel := context.WithCancel(context.Background())
 	defer ctxCancel()
 
-	// retrieve the state and make sure everything looks correct
+	// try a few times and wait around until we see the build is complete, or fail.
+	success := false
 	var grpcState *proto.State
-	attempts = 0
+	for i := 0; i < stateRetries; i++ {
+		grpcState = retrieveRPCState(ctx, t, client)
+		if checkState(*grpcState) {
+			success = true
+			break
+		}
+		time.Sleep(waitTime)
+	}
+	if !success {
+		// max attempts exceeded, log errors
+		for _, v := range grpcState.BuildState.Artifacts {
+			testutil.CheckDeepEqual(t, event.Complete, v)
+		}
+		testutil.CheckDeepEqual(t, event.Complete, grpcState.DeployState.Status)
+	}
+}
+
+func retrieveRPCState(ctx context.Context, t *testing.T, client proto.SkaffoldServiceClient) *proto.State {
+	var grpcState *proto.State
+	var err error
+	attempts := 0
 	for {
```

Comment:
instead of infinite for loop + continue + t.Fatalf + break - a more natural version would be a for loop on i:=0 to `connectionRetries` and return with a successful connection, fail after the loop...